### PR TITLE
Allow for building multi-arch docker images via buildx

### PR DESCRIPTION
**Pull request by mark-vieira**
We previously simply forbade building of Docker images for anything but the current platform. However, you can build Docker images for platforms other than the current platform via [`buildx`](https://docs.docker.com/build/buildx/multiplatform-images/). This pull request adds support for this, detecting if the host system supports alternative target platforms and then unlocking the Docker tasks for those platforms.

This is mainly being added for CI, so that we can simplify our release process and build all release artifacts on the same machine.

### CODE REVIEW

File: distribution/docker/build.gradle [hunk 1]
```diff
@@ -3,10 +3,9 @@ import org.elasticsearch.gradle.LoggedExec
 import org.elasticsearch.gradle.VersionProperties
 import org.elasticsearch.gradle.internal.DockerBase
 import org.elasticsearch.gradle.internal.distribution.InternalElasticsearchDistributionTypes
-import org.elasticsearch.gradle.internal.docker.DockerBuildTask
-import org.elasticsearch.gradle.internal.docker.ShellRetry
-import org.elasticsearch.gradle.internal.docker.TransformLog4jConfigFilter
+import org.elasticsearch.gradle.internal.docker.*
```
Comment (pugnascotia) [inline distribution/docker/build.gradle:6]: Wildcard import.

File: distribution/docker/build.gradle [hunk 2]
```diff
@@ -333,11 +332,13 @@ private static List<String> generateTags(DockerBase base) {
     namespace += '-ci'
   }
 
-  return [
-    "${image}:test",
-    "${image}:${version}",
-    "docker.elastic.co/${namespace}/${image}:${version}"
-  ]
+  def tags = ["${image}:${architecture.classifier}"]
```
Comment (pugnascotia) [inline distribution/docker/build.gradle:335]: I feel like this should include a version as well? Or is that not necessary?

**Review by pugnascotia — APPROVED** [review]
LGTM.

File: distribution/docker/build.gradle (hunk 2 above)
Comment (mark-vieira) [reply distribution/docker/build.gradle:339]: Nope, same reason why it's fine to just use `elasticsearch:test` as these are only used within the scope of a single build invocation.

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/part-3

**Pull request merged**